### PR TITLE
[fix](bitmap) fix bitmap_count errors to set nullable to non-nullable bitmap col

### LINKED ISSUE #17841: [Bug] [daily] bitmap_count() cause BE core
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Version

d4dc56c

### What's Wrong?

failed case: https://ci-builds.apache.org/job/Doris/job/doris_daily_enable_vectorized/lastBuild/testReport/junit/(root)/test_query_bitmap_filter/test_bitmap_filter_uncorrelated_2/

select k1 from baseall a where k4 in (select bm_k4 from test_k1_bitmap b where bitmap_count(bm_k4) > 250);
image

core file:

*** Query id: 76e572b376c14d27-974a0caad121db06 ***
*** Aborted at 1678936489 (unix time) try "date -d @1678936489" if you are using GNU date ***
*** Current BE git commitID: d4dc56c ***
*** SIGSEGV address not mapped to object (@0x18) received by PID 88249 (TID 88703 OR 0x7f4e4a9cc700) from PID 24; stack trace: ***
 0# doris::signal::(anonymous namespace)::FailureSignalHandler(int, siginfo_t*, void*) at /root/doris-release/be/src/common/signal_handler.h:413
 1# 0x00007F4F56565500 in /lib64/libpthread.so.0
 2# doris::vectorized::DataTypeBitMap::get_uncompressed_serialized_bytes(doris::vectorized::IColumn const&, int) const at /root/doris-release/be/src/vec/data_types/data_type_bitmap.cpp:34
 3# doris::vectorized::DataTypeNullable::get_uncompressed_serialized_bytes(doris::vectorized::IColumn const&, int) const at /root/doris-release/be/src/vec/data_types/data_type_nullable.cpp:101
 4# doris::vectorized::Block::serialize(int, doris::PBlock*, unsigned long*, unsigned long*, doris::segment_v2::CompressionTypePB, bool) const at /root/doris-release/be/src/vec/core/block.cpp:769
 5# doris::vectorized::VDataStreamSender::serialize_block(doris::vectorized::Block*, doris::PBlock*, int) at /root/doris-release/be/src/vec/sink/vdata_stream_sender.cpp:644
 6# doris::vectorized::VDataStreamSender::send(doris::RuntimeState*, doris::vectorized::Block*, bool) at /root/doris-release/be/src/vec/sink/vdata_stream_sender.cpp:512
 7# doris::PlanFragmentExecutor::open_vectorized_internal() at /root/doris-release/be/src/runtime/plan_fragment_executor.cpp:312
 8# doris::PlanFragmentExecutor::open() at /root/doris-release/be/src/runtime/plan_fragment_executor.cpp:242
 9# doris::FragmentExecState::execute() at /root/doris-release/be/src/runtime/fragment_mgr.cpp:229
10# doris::FragmentMgr::_exec_actual(std::shared_ptr<doris::FragmentExecState>, std::function<void (doris::RuntimeState*, doris::Status*)> const&) at /root/doris-release/be/src/runtime/fragment_mgr.cpp:494
11# std::_Function_handler<void (), doris::FragmentMgr::exec_plan_fragment(doris::TExecPlanFragmentParams const&, std::function<void (doris::RuntimeState*, doris::Status*)> const&)::{lambda()#1}>::_M_invoke(std::_Any_data const&) at /var/local/ldb-toolchain/include/c++/11/bits/std_function.h:291
12# doris::ThreadPool::dispatch_thread() at /root/doris-release/be/src/util/threadpool.cpp:537
13# doris::Thread::supervise_thread(void*) at /root/doris-release/be/src/util/thread.cpp:454
14# start_thread in /lib64/libpthread.so.0
15# clone in /lib64/libc.so.6

### What You Expected?

query execute success

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by luozenglin**
# Proposed changes

Issue Number: close #17841

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by luozenglin** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 35.01 seconds
 stream load tsv:          433 seconds loaded 74807831229 Bytes, about 164 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          60 seconds loaded 1101869774 Bytes, about 17 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230414142553_clickbench_pr_129271.html

File: be/src/vec/functions/function_bitmap.cpp [hunk 1]
```diff
@@ -412,7 +412,7 @@ class FunctionBitmapCount : public IFunction {
         auto data_null_map = ColumnUInt8::create(input_rows_count, 0);
         auto& null_map = data_null_map->get_data();
 
-        auto& column = block.get_by_position(arguments[0]).column;
```
Comment (yiguolei) [inline be/src/vec/functions/function_bitmap.cpp:415]: why could not use reference here?

Comment (luozenglin) [reply be/src/vec/functions/function_bitmap.cpp:415]: Because in row 418, the `column` may be assigned to a non-nullable column

**Review by zhangstar333 — APPROVED** [review]

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**